themes/edx.org/lms/templates/header.html

```diff
@@ -69,10 +76,9 @@ <h2 class="course-header">
 
     % if user.is_authenticated():
       % if not course or disable_courseware_header:
-        % if not nav_hidden or show_program_listing:
           <nav aria-label="Main" class="nav-main">
             <ul class="left list-inline nav-global authenticated">
-              % if not nav_hidden:
+              % if not nav_hidden and not show_program_listing:
                 <%block name="navigation_global_links_authenticated">
```

Comment:
@tasawernawaz From what I understand what you said, I think logically this condition is correct.
That said, the if statement is not understandable.
What about making a change to this:
`if show_program_listing:`
`  Do program Listing Links`
' else if not nav_hidden:`
`  Do "How it works" and "Schools & Partners" links`

Do you think the above logic is the same as here and more readable?